### PR TITLE
Fix after_commit on: :destroy not being triggered

### LINKED ISSUE #217: `after_commit on: :destroy` hooks are broken in 2.1.0
https://github.com/radar/paranoia/commit/2b0db84f38a0e436abbb129c770b0dac9275a802 broke `after_commit on: :destroy` callbacks because rails checks `destroyed?` after running a transaction to determine if it should run those callbacks: https://github.com/rails/rails/blob/4-2-stable/activerecord/lib/active_record/transactions.rb#L408

@huoxito @jhawthorn

This effectively makes this gem unusable starting with 2.1.0 :(

### PR DESCRIPTION
**Pull request by npezza93**
Shoutout to @skorfmann on #228 for for most of this! Fixes #217 

Let me know if there are any issues with this approach.

### CODE REVIEW

File: lib/paranoia.rb [hunk 1]
```diff
@@ -193,6 +193,24 @@ def timestamp_attributes_with_current_time
...
 
+  def transaction_include_any_action?(actions) #:nodoc:
+    actions.any? do |action|
+      case action
+      when :create
+        transaction_record_state(:new_record)
+      when :destroy
+        transaction_include_destroy?
+      when :update
+        !(transaction_record_state(:new_record) || transaction_include_destroy?)
```
Comment (BenMorganIO) [inline lib/paranoia.rb]: ~Why not just use a `x && y` instead of a `!(x || y)`?~ nvm, I had some of the boolean logic mixed up in my head...

**Comment by npezza93** [comment]
Remade this to accommodate rails 5.1 `@_trigger_{action}_callback` instance methods from this commit: https://github.com/rails/rails/commit/371c083fb0620efebf7bd0188a66486403e12ecc

**Comment by rmachielse** [comment]
@npezza93 @BenMorganIO any idea on when this will be merged? We could use the change quite well.

**Comment by npezza93** [comment]
@rmachielse ¯\\\_(ツ)\_/¯, I haven't heard anything so I think it's up to @BenMorganIO.

**Comment by rmachielse** [comment]
@BenMorganIO what are your thoughts about this?

**Comment by rmachielse** [comment]
@BenMorganIO is there any chance this gets merged soon?

**Comment by rmachielse** [comment]
@BenMorganIO anything I can do to speed this up?

**Comment by rmachielse** [comment]
Is there anyone else who can pick this up? @jhawthorn maybe?

**Comment by fenak** [comment]
That'd be very helpful for me as well, would like to see it on upstream. 👍

**Comment by koenpunt** [comment]
While this probably is the best approach; it differs from how it behaves with Rails 4.2. There a "soft delete" would trigger the update callback, and the destroy callback is called only for `really_destroy`

**Comment by dholdren** [comment]
any update on this? I'm running into this now and surprised it hasn't been addressed yet

**Comment by JorgeGarciaxyz** [comment]
Any update ?

**Comment by ruubyy** [comment]
+1

**Comment by razum2um** [comment]
@npezza93 thanks for initial PR ❤️

@BenMorganIO will you merge & release it if we'll rebase && fix travis? /cc @jhawthorn

**Comment by npezza93** [comment]
I believe this is has been fixed with a combo of the latest release and rails >= 5.2

**Pull request closed without merging**